pkg/e2e/operators/cloudingress/public_private.go

```diff
@@ -2,35 +2,128 @@ package cloudingress
 	"github.com/openshift/osde2e/pkg/common/constants"
 	"github.com/openshift/osde2e/pkg/common/helper"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 // tests
 var _ = ginkgo.Describe(constants.SuiteInforming+TestPrefix, func() {
 	h := helper.New()
 
 	ginkgo.Context("publishingstrategy-public-private", func() {
-		ginkgo.It("is a placeholder", func() {
-			_, _ = h.Kube().CoreV1().Pods(TestPrefix).Get(context.TODO(), "something", metav1.GetOptions{})
+		ginkgo.It("should be able to toggle the default applicationingress from public to private", func() {
 
-			tests := []struct {
-				Name string
-			}{}
+			updateApplicationIngress(h, "internal")
 
-			for _, test := range tests {
-				Expect(test.Name).To(Equal(""))
-			}
+			// Wait for the router-default service to have the correct annotation
+			err := pollAppIngressService(h, "router-default", "openshift-ingress", "private")
+			Expect(err).NotTo(HaveOccurred())
+		})
+		ginkgo.It("should be able to toggle the default applicationingress from private to public", func() {
+			updateApplicationIngress(h, "external")
+
+			// Wait for the router-default service to have the correct annotation
+			err := pollAppIngressService(h, "router-default", "openshift-ingress", "public")
+			Expect(err).NotTo(HaveOccurred())
 		})
 	})
 })
 
 // utils
 
+func updateApplicationIngress(h *helper.H, lbscheme string) {
+	var PublishingStrategyInstance cloudingressv1alpha1.PublishingStrategy
+
+	// Check that the PublishingStrategy CR is present
+	ps, err := h.Dynamic().Resource(schema.GroupVersionResource{Group: "cloudingress.managed.openshift.io", Version: "v1alpha1", Resource: "publishingstrategies"}).Namespace(OperatorNamespace).Get(context.TODO(), "publishingstrategy", metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(ps.Object, &PublishingStrategyInstance)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Grab the current list of Application Ingresses from the Publishing Strategy
+	AppIngress := PublishingStrategyInstance.Spec.ApplicationIngress
+
+	// Find the default router and update its scheme
+	for i, v := range AppIngress {
+		if v.Default == true {
+			AppIngress[i].Listening = cloudingressv1alpha1.Listening(lbscheme)
+		}
+	}
+
+	PublishingStrategyInstance.Spec.ApplicationIngress = AppIngress
+
+	ps.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(&PublishingStrategyInstance)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Update the publishingstrategy
+	ps, err = h.Dynamic().Resource(schema.GroupVersionResource{Group: "cloudingress.managed.openshift.io", Version: "v1alpha1", Resource: "publishingstrategies"}).Namespace(OperatorNamespace).Update(context.TODO(), ps, metav1.UpdateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func pollAppIngressService(h *helper.H, serviceName string, svcNamespace string, lbscheme string) error {
```

Comment:
When doing polls, there are helper funcs for that:
https://github.com/openshift/osde2e/blob/d9e539bbd336ed12dc0e74d03f22b700fdf14e79/pkg/e2e/operators/cloudingress/apischeme_cr.go#L26-L34
https://pkg.go.dev/k8s.io/apimachinery@v0.20.2/pkg/util/wait#PollImmediate

Reply:
Great! Didn't know about those. I've updated the tests to use them and removed the `pollAppIngressService` function.